internal/transport/http2_server.go

```diff
@@ -849,6 +849,9 @@ func (t *http2Server) WriteStatus(s *Stream, st *status.Status) error {
 func (t *http2Server) Write(s *Stream, hdr []byte, data []byte, opts *Options) error {
 	if !s.isHeaderSent() { // Headers haven't been written yet.
 		if err := t.WriteHeader(s, nil); err != nil {
+			if err == ErrConnClosing {
```

Comment:
Why only for ErrConnClosing and not for every possible ConnectionError?
```suggestion
			if _, ok := err.(ConnectionError); ok {
```